plugins/inputs/kafka_consumer/kafka_consumer.go

```diff
@@ -225,6 +257,80 @@ func (k *KafkaConsumer) Stop() {
 	k.wg.Wait()
 }
 
+func (k *KafkaConsumer) addTopicRegexps() error {
+	// We instantiate a new generic Kafka client, so we can ask
+	// it for all the topics it knows about.  Then we build
+	// regexps from our strings, loop over those, loop over the
+	// topics, and if we find a match, add that topic to
+	// out topic set, which then we turn back into a list at the end.
+	var err error
+	var allTopics []string
+	var exists = struct{}{}
+	regexps := make([]regexp.Regexp, 0, len(k.TopicRegexps))
+
+	kafkaClient, err := sarama.NewClient(k.Brokers, k.config)
+	if err != nil {
+		return err
+	}
+	defer kafkaClient.Close()
+	allTopics, err = kafkaClient.Topics()
+	if err != nil {
+		return err
+	}
+	for _, r := range k.TopicRegexps {
+		re := regexp.MustCompile(r)
+		regexps = append(regexps, *re)
+	}
+	topicSet := make(map[string]struct{})
+	for _, t := range k.Topics {
+		// Get our preexisting topics
+		topicSet[t] = exists
+	}
+	var removeTopics []string
+	for k := range topicSet {
+		found := false
+		for _, t := range allTopics {
+			if t == k {
+				found = true
+				break
+			}
+		}
+		if !found {
+			removeTopics = append(removeTopics, k)
+		}
+	}
+	// Remove any topics no longer present on broker
+	for _, w := range removeTopics {
+		k.Log.Infof("topic '%s' no longer present; removing", w)
+		delete(topicSet, w)
+	}
+	// Add topics that have newly appeared on broker
+	for _, r := range regexps {
+		for _, t := range allTopics {
+			if r.MatchString(t) {
+				if _, ok := topicSet[t]; !ok {
```

Comment:
We can probably save one level of indentation here
```suggestion
			if _, ok := topicSet[t]; !ok && r.MatchString(t) {
```